From b8f4b899b4be67ebe18901e9cd570415b4b5ab83 Mon Sep 17 00:00:00 2001
From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Wed, 22 Feb 2017 17:32:08 -0600
Subject: [PATCH 1/4] set_flags_from_control: remove function

During installs, install_data_files calls set_flags_from_control to
capture the case where fields, such as "Essential", are present in the
package control file but are missing on the Packages file. This
operation is expensive (re-parses the control file) and is catering to a
case that should not be supported on the first place.

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>
---
 libopkg/opkg_install.c |  7 -------
 libopkg/pkg.c          | 28 ----------------------------
 2 files changed, 35 deletions(-)

diff --git a/libopkg/opkg_install.c b/libopkg/opkg_install.c
index f021197..9a564a9 100644
--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -768,13 +768,6 @@ static int install_data_files(pkg_t * pkg)
         return err;
     }
 
-    /* The "Essential" control field may only be present in the control
-     * file and not in the Packages list. Ensure we capture it regardless.
-     *
-     * XXX: This should be fixed outside of opkg, in the Package list.
-     */
-    set_flags_from_control(pkg);
-
     opkg_msg(DEBUG, "Calling pkg_write_filelist.\n");
     err = pkg_write_filelist(pkg);
     if (err)
diff --git a/libopkg/pkg.c b/libopkg/pkg.c
index 0f8e454..c7230a3 100644
--- a/libopkg/pkg.c
+++ b/libopkg/pkg.c
@@ -430,34 +430,6 @@ abstract_pkg_t *abstract_pkg_new(void)
     return ab_pkg;
 }
 
-void set_flags_from_control(pkg_t * pkg)
-{
-    char *file_name;
-    FILE *fp;
-    int r;
-
-    sprintf_alloc(&file_name, "%s/%s.control", pkg->dest->info_dir, pkg->name);
-
-    fp = fopen(file_name, "r");
-    if (fp == NULL) {
-        opkg_perror(ERROR, "Failed to open %s", file_name);
-        free(file_name);
-        return;
-    }
-
-    free(file_name);
-
-    r = pkg_parse_from_stream(pkg, fp, PFM_ALL ^ PFM_ESSENTIAL);
-    if (r != 0) {
-        opkg_msg(DEBUG, "Unable to read control file for %s. May be empty.\n",
-                 pkg->name);
-    }
-
-    fclose(fp);
-
-    return;
-}
-
 static const char *pkg_state_want_to_str(pkg_state_want_t sw)
 {
     unsigned int i;
-- 
2.7.4

